### PR TITLE
docs: adjust README.md and CONTRIBUTING.md

### LINKED ISSUE #275: Add reference to the GitHub pages in README.md

### PR DESCRIPTION
**Pull request by behnazh-w**
Closes #275 and #272

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -58,19 +58,130 @@ can be accepted.
 
 - Each new feature or change PR should provide meaningful CI tests.
 - All the tests and security scanning checks in CI should pass and achieve the expected coverage.
+- To avoid unnecessary failures in GitHub Actions, make sure `make check` and `make test` work locally.
+See below for instructions to set up the development environment.
 
 ### Merging PRs
 
-- Before a PR is merged, all commits in the PR should be rebased into meaningful commits.
+- Before a PR is merged, all commits in the PR should meaningful commits.
```
Comment (tromai) [inline CONTRIBUTING.md:66]:
```suggestion
- Before a PR is merged, all commits in the PR should have meaningful commits.
```
We could use another word if it make more sense.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -58,19 +58,130 @@ can be accepted.
...
-  - This ensures a linear commit history for the project.
-  - If the fast-forward merge does not work due to new commits on `main` (in this case the two branches diverge and the PR branch cannot be fast-forwarded), the PR branch should be rebased onto `main` first.
+- PRs should be merged using the `Squash and merge` strategy. In most cases a single commit with
+a detailed commit message body is preferred. Make sure to keep the `Signed-off-by` line in the body.
 
 ## Branching model
 
 * The `main` branch is only used for releases and the `staging` branch is used for development. We only merge to `main` when we want to create a new release for Macaron.
 
+## Setting up the development environment
```
Comment (tromai) [inline CONTRIBUTING.md:75]: Could you rebase on staging and update the reference to this section in `docs/README.md` - https://github.com/oracle-samples/macaron/blob/e6656c486f956e843e250d8dec3ace0bf36a2ecb/docs/README.md?plain=1#L7

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -58,19 +58,130 @@ can be accepted.
...
+```
+
+### Running Macaron as a Python package
+
+```bash
+usage: macaron [-h]
+```
+
+### Obtaining the GitHub personal access token
+Create your own Github access token (please refer to the instructions [here](https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token)). When creating this token, make sure to assign **at least** `repo` permissions.
```
Comment (tromai) [inline CONTRIBUTING.md:128]: I think we can update the instruction on permission of the Github token by using parts of the instructions we have for the Sphinx documentation

> Ideally, the GitHub token must have read permissions for the repositories that you want to analyze:
> 
> - Every [fine-grained personal-access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-fine-grained-personal-access-token) should have read permission to public GitHub repositories. However, if you are analyzing a private repository, please select it in the Repository Access section.
> - For [classic personal-access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-personal-access-token-classic), the repo.public_repo scope must be selected. Please select the whole repo scope if you are running the analysis against private repositories.
>

File: CONTRIBUTING.md (hunk 1 above)
Comment (behnazh-w) [reply CONTRIBUTING.md]: See commit [6de60e8](https://github.com/oracle-samples/macaron/pull/279/commits/6de60e887c592f5ef3b4bb984985148507222b1f)

File: CONTRIBUTING.md (hunk 2 above)
Comment (behnazh-w) [reply CONTRIBUTING.md:75]: See commit [6de60e8](https://github.com/oracle-samples/macaron/pull/279/commits/6de60e887c592f5ef3b4bb984985148507222b1f)

File: CONTRIBUTING.md (hunk 3 above)
Comment (behnazh-w) [reply CONTRIBUTING.md]: See commit [6de60e8](https://github.com/oracle-samples/macaron/pull/279/commits/6de60e887c592f5ef3b4bb984985148507222b1f)

**Review by tromai — APPROVED** [review]

**Pull request merged**